process/block/metablock.go

```diff
@@ -455,13 +469,20 @@ func (mp *metaProcessor) CommitBlock(
 		return err
 	}
 
+	numberOfShards := mp.shardCoordinator.NumberOfShards()
+	headersNonce := make([]uint64, numberOfShards, numberOfShards)
```

Comment:
second numberOfShards is useless here

Reply:
Removed headersNonce slice and added a new sync map to store all  headersNonce.